Albania/ALB_extract_raw_microdata_excel_to_csv.py

```diff
@@ -76,76 +76,76 @@ def pad_left(code, length = 3):
     while len(code)<length:
         code = '0'+code
     return code
+        
+def format_float(x):
+    if pd.isna(x):
+        return x
+    x = str(x).strip()
+    try:
+        return float(x)
+    except ValueError:
+        pass
+    euro_thousands_pattern = r'^\d{1,3}(\.\d{3})+,\d{2}$'
+    if re.match(euro_thousands_pattern, x):
+        try:
+            x = x.replace('.', '').replace(',', '.')
+            return float(x)
+        except ValueError:
+            return pd.NA
+    comma_decimal_pattern = r'^\d+,\d{1,2}$'
+    if re.match(comma_decimal_pattern, x):
+        try:
+            x = x.replace(',', '.')
+            return float(x)
+        except ValueError:
+            return pd.NA
+    return pd.NA
 
 # COMMAND ----------
 
-col_names = {
-    'Gov Entity':'admin2',
-    'Gov. Entity':'admin2',
-    'Gov.Entity':'admin2',
-    'Gov':'admin2',
-    'ge':'admin2',
-    'Line Ministri':'admin3',
-    'Line Ministry':'admin3',
-    'Ministry':'admin3',
-    'lm':'admin3',
-    'Inst.':'admin4',
-    'Institution':'admin4',
-    'institution': 'admin4',
-    'Chapter':'fin_source',
-    'ch':'fin_source',
-    'program':'func3',
-    'Program':'func3',
-    'Progr':'func3',
-    'Account':'econ5',
-    'Economic Account':'econ5',
-    'eccaccount':'econ5',
-    'TDO':'admin5',
-    'tdo':'admin5',
-    'FYTD Actual':'executed',
-    'actual':'executed',
-    'Actual':'executed',
-    'Operational Budget':'revised',
-    'operationalbudget':'revised',
-    'Initial Budget':'approved',
-    'initialbudget':'approved',
-    'Project':'project',
-    'Account Description': 'Account_Description',
-    'Institution Description':'Institution_Description'
-
-}
-sheet_map = {
-    ('3 digit', 2023): ('Vendori', 0),
-    ('3 digit', 2024): ('Sheet2', 1),
-}
-
-required_col_names = [
-    'admin2', 'admin3', 'admin4', 'admin5', 'fin_source', 'func3', 'econ5', 'project', 'executed', 'revised', 'approved'
-    ]
+col_names_3_digit = [
+    'admin2', 'admin3', 'admin4', 'fin_source', 'func3', 'econ3', 'admin5', 'project', 'executed', 'revised', 'approved']
+col_names_7_digit = [
+    'admin2', 'admin3', 'admin4', 'fin_source', 'func3', 'econ5', 'admin5', 'project', 'executed']
```

Comment:
@bhupatiraju I thought we discussed and agreed – now that we are assuming that the columns come in the fixed order, we should do some sanity checks through type / format / length assertion to roughly check that each of these columns are as expected. I don't see the assertions below. Can we add?